### PR TITLE
[VDO-5602] dm vdo: Restyle function declarations, definition headers, and calls to align with dm code style

### PR DESCRIPTION
**Pull request by raeburn**
Mostly automated wrapping of function declarations and function calls, with some manual patching up, based on some feedback and patches from Mike Snitzer.

### CODE REVIEW

**Comment by raeburn** [comment]
Per Matt's request, hold off on this untli some overlapping changes of his are committed

File: src/c++/uds/src/uds/volume.c [hunk 1]
```diff
@@ -1411,8 +1349,8 @@ static void probe_chapter(struct volume *volume, u32 chapter_number, u64 *virtua
 
 #endif /* TEST_INTERNAL*/
 	*virtual_chapter_number = BAD_CHAPTER;
-	dm_bufio_prefetch(volume->client,
-			  map_to_physical_page(geometry, chapter_number, 0),
+	dm_bufio_prefetch(volume->client, map_to_physical_page(geometry, chapter_number,
+							       0),
 			  geometry->index_pages_per_chapter);
```
Comment (lorelei-sakai) [inline src/c++/uds/src/uds/volume.c:1354]: This wrapping is actively worse. Please revert.

File: src/c++/uds/src/uds/volume.c [hunk 2]
```diff
@@ -1649,33 +1578,27 @@ initialize_page_cache(struct page_cache *cache,
 	atomic64_set(&cache->clock, 1);
 
 	result = ASSERT((cache->cache_slots <= VOLUME_CACHE_MAX_ENTRIES),
-			"requested cache size, %u, within limit %u",
-			cache->cache_slots,
+			"requested cache size, %u, within limit %u", cache->cache_slots,
 			VOLUME_CACHE_MAX_ENTRIES);
```
Comment (lorelei-sakai) [inline src/c++/uds/src/uds/volume.c:1582]: Flowing this 1/1/2 so the format string is by itself would be clearer.

File: src/c++/uds/src/uds/volume.h [hunk 3]
```diff
@@ -135,23 +135,20 @@ int __must_check uds_replace_volume_storage(struct volume *volume,
...
 int __must_check uds_find_volume_chapter_boundaries(struct volume *volume,
-						    u64 *lowest_vcn,
-						    u64 *highest_vcn,
+						    u64 *lowest_vcn, u64 *highest_vcn,
 						    bool *is_empty);
 
-int __must_check
-uds_search_volume_page_cache(struct volume *volume, struct uds_request *request, bool *found);
+int __must_check uds_search_volume_page_cache(struct volume *volume, struct uds_request *request,
+					      bool *found);
```
Comment (lorelei-sakai) [inline src/c++/uds/src/uds/volume.h:143]: Though not technically too long, I think looks better flowed 1/1/1.

File: src/c++/uds/src/uds/volume.h [hunk 4]
```diff
@@ -161,17 +158,15 @@ int __must_check uds_write_chapter(struct volume *volume,
...
-				   struct delta_index_page index_pages[]);
+int __must_check uds_read_chapter_index_from_volume(const struct volume *volume,
+						    u64 virtual_chapter,
+						    struct dm_buffer *volume_buffers[],
+						    struct delta_index_page index_pages[]);
 
-int __must_check
-uds_get_volume_record_page(struct volume *volume, u32 chapter, u32 page_number, u8 **data_ptr);
+int __must_check uds_get_volume_record_page(struct volume *volume, u32 chapter, u32 page_number,
+					    u8 **data_ptr);
```
Comment (lorelei-sakai) [inline src/c++/uds/src/uds/volume.h:168]: nearly too long. Would look better at 2/2.

**Review by lorelei-sakai — CHANGES_REQUESTED** [review]
Wow I had a whole comment in here I lost. I'll just say, 211 comments, but only about four or five classes of common things, which we mostly talked about. There are enough brace and newline issues that I noticed that I'll put out another commit supplemental to my earlier one rather than try to absorb all those thing here.

Please especially note the comments where Mike's proposed changes differ from yours. I am very sorry this has turned into so much work.

File: src/c++/vdo/base/block-map.c [hunk 5]
```diff
@@ -2633,7 +2595,8 @@ static void traverse(struct cursor *cursor)
```
[the remark this answers is not in the document]
Comment (lorelei-sakai) [reply src/c++/vdo/base/block-map.c:2595]: [stolen]

File: src/c++/vdo/base/data-vio.c [hunk 6]
```diff
@@ -808,8 +804,7 @@ static void process_release_callback(struct vdo_completion *completion)
 		wake_up_nr(&pool->limiter.blocked_threads, to_wake);
 
 	if (discards_to_wake > 0) {
-		wake_up_nr(&pool->discard_limiter.blocked_threads,
-			   discards_to_wake);
+		wake_up_nr(&pool->discard_limiter.blocked_threads, discards_to_wake);
 	}
```
Comment (lorelei-sakai) [inline src/c++/vdo/base/data-vio.c:808]: Putting a flag here so I remember to look at it again. This is the place where braces need to go away, now that the block is just a single line.

File: src/c++/vdo/base/slab-depot.c [hunk 7]
```diff
@@ -5143,8 +4985,7 @@ void vdo_scrub_all_unrecovered_slabs(struct slab_depot *depot, struct vdo_comple
  *
```
Comment (lorelei-sakai) [inline src/c++/vdo/base/slab-depot.c:4985]: This is an unrelated thing I'd like you to roll in. Coud you fix the doc for get_block_allocator_statistics() to match upstream? That is, newline after "allocators" (not before), and align the second line correctly?

This is a place where somehow vdo-devel is out of step with our upstream baseline, I'd like to align them again.

**Review by lorelei-sakai — APPROVED** [review]
It's been a long road, But this all looks good to me now.

**Pull request merged**